File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/Subscription.py
from typing import Optional, List
from pydantic import *
from ..base.ContactPoint import ContactPoint
from ..base.DomainResource import DomainResource
from ..base.BackboneElement import BackboneElement

class Subscription_Channel(BackboneElement):
    type: str
    header: Optional[List[str]] = None
    payload: Optional[str] = None
    endpoint: Optional[str] = None

class Subscription(DomainResource):
    end: Optional[str] = None
    error: Optional[str] = None
    reason: str
    status: str
    channel: Subscription_Channel
    contact: Optional[List[ContactPoint]] = None
    criteria: str

File: test/aidbox_sdk/snapshots/typescript/hl7-fhir-r4-core/Subscription.ts
import { ContactPoint } from "./ContactPoint";
import { DomainResource } from "./DomainResource";
import { BackboneElement } from "./BackboneElement";

export type SubscriptionChannel = BackboneElement & {
    type: string;
    header?: string[];
    payload?: string;
    endpoint?: string;
};

export type Subscription = DomainResource & {
    end?: string;
    error?: string;
    reason: string;
    status: string;
    channel: SubscriptionChannel;
    contact?: ContactPoint[];
    criteria: string;
};